sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveTableProperties.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.catalyst.analysis
+
+import scala.collection.JavaConverters._
+
+import org.apache.spark.sql.AnalysisException
+import org.apache.spark.sql.catalyst.plans.logical.{AlterTableUnsetProperties, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+import org.apache.spark.sql.connector.catalog.CatalogV2Implicits.IdentifierHelper
+import org.apache.spark.sql.connector.catalog.TableCatalog
+
+/**
+ * A rule for resolving AlterTableUnsetProperties to handle non-existent properties.
+ */
+object ResolveTableProperties extends Rule[LogicalPlan] {
+  def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
+    case a @ AlterTableUnsetProperties(r: ResolvedTable, props, ifExists) if !ifExists =>
+      val tblProperties = r.table.properties.asScala
+      props.foreach { p =>
+        if (!tblProperties.contains(p) && p != TableCatalog.PROP_COMMENT) {
```

Comment:
Just found a problem here: removing table property may be case insensitive (depends on the v2 table implementation), so the code here may break existing queries unexpectedly.

I think the right approach is to change the v2 API and pass the `ifExists` flag to the v2 implementations. However, I feel like the current v2 behavior is OK (do not fail even if the property doesn't exist). So another idea is to change the v1 behavior instead, and ignore the `ifExists` flag and always do not fail.

@imback82 @rdblue what do you think?

Reply:
Good catch. I don't have a strong opinion here; both approaches sound reasonable to me, each with pros/cons.

Reply:
Let me revert this commit first. We can start with the "change v1 to match v2" idea, as it's less breaking (some failed commands become noop).

Reply:
If we want to go with a less breaking change, isn't updating V2 API a better option (you can check the change [here](https://github.com/apache/spark/pull/31494/commits/0065170f24a2ff00cfee920e8fd31b0574f55ac7#diff-ca6d2813b4735973d1fb343a7b19271074915ce44590445899a3e2ea88ae744eR78)) since we are just adding a new API?

The only concern I have with "change v1 to match v2" idea is that users expect the behavior would be different if there exists an `IF EXISTS` option.

Reply:
Got it. Thank you, @cloud-fan .

Reply:
Code-wise, yes, adding a new field to `RemoveProperty` is less breaking as it's binary compatible. However, we also changed the semantic of `RemoveProperty` and force all v2 implementations to update and respect the new `ifExists` flag, which is very breaking.

I'd like to also point out that, the current v2 behavior is idempotent, you can run `UNSET TBLPROPERTIES` more than once and the result won't change. It's a good property and is consistent with `SET TBLPROPERTIES`.

Reply:
OK, makes sense. Thanks.

Reply:
I think that the new `ifExists` property in `RemoveProperty` would be easily overlooked or ignored. In general, I prefer fixing these problems in Spark and not adding requirements to the sources that will choose whether or not to follow them.

Reply:
Can we directly uses `r.table.properties.containsKey` ? In that way the `Table` implementation side returns a case (in)sensitive map via `properties` for case sensitivity. If they want them to treat table properties in a case insensitive way, I think it makes sense to return a case insensitive map from `Table.properties`.

Reply:
@HyukjinKwon then it will be a new requirement for the v2 implementations and is a breaking change. It also means we push more work to the v2 implementation side and is not good.